core/frontend/src/Viewport.ts

```diff
@@ -1304,6 +1304,32 @@ export abstract class Viewport implements Disposable, TileUser {
     removals.push(settings.onTimePointChanged.addListener(scheduleChanged));
     removals.push(style.onScheduleScriptChanged.addListener(scriptChanged));
 
+
+    const scheduleEditingChanged = (
+      changes: Array<{ timeline: RenderSchedule.ModelTimeline, elements: Set<Id64String> }>
+    ) => {
+      for (const { timeline, elements } of changes) {
+        for (const ref of this.getTileTreeRefs()) {
+          const tree = ref.treeOwner.tileTree;
+          if (tree instanceof IModelTileTree && tree.modelId === timeline.modelId) {
```

Comment:
Add this type to `RenderSchedule` namespace:
```
export interface EditingChanges {
  timeline: RenderSchedule.ModelTimeline;
  elements: Set<Id64String>;
}
```
Use that instead of inlining the type everywhere.

Define new `@internal` method `TileTree.onScheduleEditingChanged(changes: RenderSchedule.EditingChanges[]): void` that does nothing by default. Call it here instead of looping and casting. Override it in `IModelTileTree`.

Then you will have a path forward to make V4 tiles support this workflow.